src/resampling.jl

```diff
@@ -192,21 +203,28 @@ julia> d = resample_czt(a, (x->1.0,x->1.0), shear=(x->50*x^2,0.0)); # a more com
 julia> @ve a,b,c,d # visualize distortions
 ```
 """
-function resample_czt(arr::AbstractArray{T,N}, rel_zoom; shear=nothing, shear_dim=nothing, fix_nyquist=false, new_size = size(arr), rel_pad=0.2, do_damp=false, center=CtrMid) where {T,N}
+function resample_czt(arr::AbstractArray{T,N}, rel_zoom; shear=nothing, shear_dim=nothing, fix_nyquist=false, new_size = size(arr), rel_pad=0.2, do_damp=false, center=CtrMid, kill_wrap=true, pad_value=zero(eltype(arr))) where {T,N}
     RT = real(T)
     orig_size = size(arr)
     if do_damp
         arr = damp_edge_outside(arr, rel_pad)
     else
         arr = copy(arr)
     end
-    for d in 1:length(rel_zoom)
+    for d in eachindex(rel_zoom)
         sd = mod(d,ndims(arr))+1
         if !isnothing(shear_dim)
             sd = shear_dim[d]
         end
         my_zoom = 1.0
```

Comment:
did not check, but does that hard coded Float64 1.0 work with Float32 arrays?
Can you check for type preservation?

Reply:
Yes. The scale factors are rounded and cast to an Integer anyway. But I now also included tests to this property.